### PR TITLE
Fixing SOLID [Dependency Inversion Violations] in use case code + renaming Main Panel of UI

### PR DESCRIPTION
**Pull request by carmen-chau**
## Description 
[ Briefly describe your changes to main in this section. ]

## Details
[ List your changes, in detail, to main in this section. ]

## Relevant issues
[ List/link any relevant issues/bugs/other PRs in this section if you haven't already done so in the above two. ]

### CODE REVIEW

File: src/use_case/search/SearchDataAccessInterface.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public interface SearchDataAccessInterface {
 
     List<String> getStationAmenities(String inputStationName);
 
-    void setStation(Station stationObj);
+    void setStation(String stationName);
     void setStationAmenities(Station stationObj);
```
Comment (jacqnnn) [inline src/use_case/search/SearchDataAccessInterface.java:22]: should we fix the parameter in setStationAmenities as String stationName or keep it the same

**Review by jacqnnn — COMMENTED** [review]
Overall looks great! :D 
Have some minor concerns regarding SOLID violations left as comment

File: src/data_access/text_file/FileStationDataAccessObject.java [hunk 2]
```diff
@@ -125,12 +125,12 @@ public List<Train> getIncomingVehicles(String inputStationName) {
     }
 
     @Override
-    public void setStation (Station stationObj) {
-        //Get station name
-        String stationName = getStationName(stationObj);
+    public void setStation (String stationName) {
+        //Get station object
+        Station stationObj = getStation(stationName);
```
[the remark this answers is not in the document]
Comment (carmen-chau) [reply src/data_access/text_file/FileStationDataAccessObject.java]: It is needed in subsequent lines, so will not be removing it

[the remark this answers is not in the document]
Comment (jacqnnn) [reply src/data_access/text_file/FileStationDataAccessObject.java]: oh yea just realized it's related to my use case, my bad. will modify it on my end after this pr is merged

File: src/data_access/text_file/FileStationDataAccessObject.java [hunk 3]
```diff
@@ -139,7 +139,8 @@ public void setStation (Station stationObj) {
     }
 
     @Override
-    public void setStationAmenities(Station stationObj){
+    public void setStationAmenities(String stationName){
+        Station stationObj = getStation(stationName);
```
[the remark this answers is not in the document]
Comment (jacqnnn) [reply src/data_access/text_file/FileStationDataAccessObject.java]: Will fix mine too

**Review by jacqnnn — APPROVED** [review]
Code looks great and works as desired. Great work woohooo :D

**Pull request merged**